### PR TITLE
RISImporterTest [v2]

### PR DESCRIPTION
**Pull request by obraliar**

### CODE REVIEW

**Comment by obraliar** [comment]
All requested test cases from https://github.com/JabRef/jabref/pull/505 have been successfully implemented.

**Comment by obraliar** [comment]
Ready to merge?

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 1]
```diff
@@ -28,82 +26,196 @@ public void setUp() throws Exception {
...
     public void testIsRecognizedFormat() throws IOException {
 
         RisImporter importer = new RisImporter();
-        try (InputStream stream = RISImporterTest.class
-                .getResourceAsStream("RisImporterTest1.ris")) {
+
+        try (InputStream stream = RISImporterTest.class.getResourceAsStream("RisImporterTest1.ris")) {
             Assert.assertTrue(importer.isRecognizedFormat(stream));
         }
+        try (InputStream stream = RISImporterTest.class.getResourceAsStream("RisImporterTest2.ris")) {
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Move to new test method

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 2]
```diff
@@ -28,82 +26,196 @@ public void setUp() throws Exception {
...
 
-        HashMap<String, String> hm = new HashMap<>();
-        hm.put("title", "/sub 3/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$_3$", hm.get("title"));
+        try (InputStream stream = RISImporterTest.class.getResourceAsStream("RisImporterTest1.ris")) {
+            List<BibEntry> entries = importer.importEntries(stream, new OutputPrinterToNull());
+            BibEntry entry = entries.get(0);
+
+            Assert.assertEquals(1, entries.size());
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Create a new `expected` bibentry and compare the parsed list against it using the methods of BibtexEntryAssert (see the recent discussion in the other importer PR)

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 3]
```diff
@@ -28,82 +26,196 @@ public void setUp() throws Exception {
...
+            Assert.assertEquals("whatever", entry.getField("doi"));
+
+        }
+    }
 
-        hm.put("title", "/sub 31/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$_{31}$", hm.get("title"));
+    @Test // Test for jabref/issues/296
+    public void testImportEntriesIssue296() throws IOException {
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Better name the test to reflect what the issue was

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 4]
```diff
@@ -28,82 +26,196 @@ public void setUp() throws Exception {
...
+        RisImporter importer = new RisImporter();
 
-        hm.put("abstract", "/sub 3/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$_3$", hm.get("abstract"));
+        try (InputStream stream = RISImporterTest.class.getResourceAsStream("RisImporterTest3.ris")) {
+            List<BibEntry> entries = importer.importEntries(stream, new OutputPrinterToNull());
+            BibEntry entry = entries.get(0);
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Same as above

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 5]
```diff
@@ -28,82 +26,196 @@ public void setUp() throws Exception {
...
+        BibtexImporter bibtexImporter = new BibtexImporter();
+
+        try (InputStream risStream = RISImporterTest.class.getResourceAsStream("RisImporterTest4.ris");
+                InputStream bibStream = RISImporterTest.class
+                        .getResourceAsStream("RisImporterTest4_comparefile.bib");) {
+
+            List<BibEntry> risEntries = risImporter.importEntries(risStream, new OutputPrinterToNull());
+            List<BibEntry> bibEntries = bibtexImporter.importEntries(bibStream, new OutputPrinterToNull());
+
+            Assert.assertEquals(risEntries.size(), bibEntries.size());
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Better use BibtexEntryAssert.assertequals method

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 6]
```diff
@@ -28,82 +26,196 @@ public void setUp() throws Exception {
...
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$^3$", hm.get("abstract"));
+        }
+    }
 
-        hm.put("review", "/sup 31/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$^{31}$", hm.get("review"));
+    @Test // Test for http://tex.stackexchange.com/questions/288511/failing-to-import-into-jabref-3-0
+    public void testImportEntriesStackExchange1() throws IOException {
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Better test name

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 7]
```diff
@@ -28,82 +26,196 @@ public void setUp() throws Exception {
...
+        try (InputStream risStream = RISImporterTest.class.getResourceAsStream("RisImporterTest5.ris");
+                InputStream bibStream = RISImporterTest.class
+                        .getResourceAsStream("RisImporterTest5_comparefile.bib");) {
+
+            List<BibEntry> risEntries = risImporter.importEntries(risStream, new OutputPrinterToNull());
+            List<BibEntry> bibEntries = bibtexImporter.importEntries(bibStream, new OutputPrinterToNull());
+
+            Assert.assertEquals(risEntries.size(), bibEntries.size());
+
+            int c = 0;
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: see above

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 8]
```diff
@@ -28,82 +26,196 @@ public void setUp() throws Exception {
...
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$_{\\$Hello}$", hm.get("title"));
+        }
     }
 
-    @Test
-    public void testImportEntries() throws IOException {
-        RisImporter importer = new RisImporter();
+    @Test // Test for jabref/issues/742
+    public void testImportEntriesIssue742() throws IOException {
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Better name

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java (hunk 1 above)
Comment (obraliar) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Done.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java (hunk 2 above)
Comment (obraliar) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Done.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java (hunk 3 above)
Comment (obraliar) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Done.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java (hunk 4 above)
Comment (obraliar) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Done.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java (hunk 5 above)
Comment (obraliar) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Done.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java (hunk 6 above)
Comment (obraliar) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Done.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java (hunk 7 above)
Comment (obraliar) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Done.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java (hunk 8 above)
Comment (obraliar) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Done.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 9]
```diff
@@ -28,82 +27,147 @@ public void setUp() throws Exception {
...
+                        .getResourceAsStream("RisImporterTest4_comparefile.bib");) {
+
+            List<BibEntry> risEntries = risImporter.importEntries(risStream, new OutputPrinterToNull());
+            List<BibEntry> bibEntries = bibImporter.importEntries(bibStream, new OutputPrinterToNull());
+
+            Assert.assertEquals(risEntries.size(), bibEntries.size());
+
+            int referenceCount = 0;
+            for (int i = 0; i < risEntries.size(); i++) {
+                for (int j = 0; j < bibEntries.size(); j++) {
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Wait until #982 is merged in, then a simpler BibtexEntryAssert.assertEquals() should work.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 10]
```diff
@@ -28,82 +27,147 @@ public void setUp() throws Exception {
...
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$^3$", hm.get("title"));
 
-        hm.put("title", "/sup 31/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$^{31}$", hm.get("title"));
+    @Test
+    public void testImportEntries() throws IOException {
+        RisImporter risImporter = new RisImporter();
+        BibtexImporter bibImporter = new BibtexImporter();
```
Comment (simonharrer) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Why do you use the BibtexImporter here as well? I would recommend to have a look at the already merged classes which do not rely on this importer for their tests.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 11]
```diff
@@ -28,82 +27,147 @@ public void setUp() throws Exception {
...
-            Assert.assertEquals("article", entry.getType());
-            Assert.assertEquals("Empirical Software Engineering", entry.getField("journal"));
-            Assert.assertEquals("2001", entry.getField("year"));
-            Assert.assertEquals("6", entry.getField("volume"));
-            Assert.assertEquals("3", entry.getField("number"));
-            Assert.assertEquals("193--194", entry.getField("pages"));
-            Assert.assertEquals("#sep#", entry.getField("month"));
+            Assert.assertEquals(1, entries.size());
+
+            Assert.assertEquals("Dermoscopy Image Analysis", entry.getField("booktitle"));
```
Comment (simonharrer) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Please compare the whole entry here as well (it does not hurt).

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 12]
```diff
@@ -28,82 +27,147 @@ public void setUp() throws Exception {
...
-        hm.put("review", "/sup 31/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$^{31}$", hm.get("review"));
+            Assert.assertEquals(1, risEntries.size());
+            Assert.assertEquals(1, bibEntries.size());
 
-        hm.put("title", "/sub $Hello/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$_{\\$Hello}$", hm.get("title"));
+            BibtexEntryAssert.assertEquals(bibEntries, risEntries);
```
Comment (simonharrer) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: I would recommend to use data driven tests here as well. See comment https://github.com/JabRef/jabref/pull/508#issuecomment-196202065

Comment (obraliar) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Done.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java (hunk 11 above)
Comment (obraliar) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Done.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java (hunk 10 above)
Comment (obraliar) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: Now we are using parametrized test class, which should be better.

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java [hunk 13]
```diff
@@ -2,108 +2,39 @@
...
-import java.util.List;
 
-/**
- * Test cases for the RISImporter
- *
- * @author $Author: coezbek $
- */
 public class RISImporterTest {
 
+    RisImporter risImporter;
```
Comment (simonharrer) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java]: make this private

File: src/test/java/net/sf/jabref/importer/fileformat/RISImporterTestFiles.java [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+@RunWith(Parameterized.class)
+public class RISImporterTestFiles {
+
+    RisImporter risImporter;
```
Comment (simonharrer) [inline src/test/java/net/sf/jabref/importer/fileformat/RISImporterTestFiles.java]: make this private

**Comment by simonharrer** [comment]
:+1: LGTM

**Comment by obraliar** [comment]
Prepare for merge (merge commits)?

**Comment by tobiasdiez** [comment]
LGTM, too.
Two last remarks:
- `RisImporterTest2.bib` is not used, right? Delete it.
- Rename `RisImporterTest2.ris` to `RisImporterCorrupted.ris` (since this what the file represents, isn't it?)

Finally squash all commit together then this can be merged in.

**Comment by obraliar** [comment]
Done.

**Pull request merged**